pkgs/development/libraries/gmp/6.x.nix

```diff
@@ -48,6 +48,10 @@ let self = stdenv.mkDerivation rec {
     ++ optional (!withStatic && stdenv.hostPlatform.isWindows) "--disable-static --enable-shared"
     ++ optional (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64) "--disable-assembly";
 
+  # Required for "undefined reference to __memset_chk" error
+  # see https://github.com/msys2/MINGW-packages-dev/pull/6
+  NIX_LDFLAGS = lib.optionalString stdenv.targetPlatform.isWindows "-lssp";
```

Comment:
`CPPFLAGS=-fstack-protector` also gets through the `__memset_chk` error, but later hits  `undefined reference to `__stack_chk_fail'`